executions/graphql-kotlin-dataloader-instrumentation/src/main/kotlin/com/expediagroup/graphql/dataloader/instrumentation/level/state/ExecutionLevelDispatchedState.kt

```diff
@@ -49,12 +48,9 @@ class ExecutionLevelDispatchedState(
      */
     fun beginExecuteOperation(
         parameters: InstrumentationExecuteOperationParameters
-    ): InstrumentationContext<ExecutionResult> {
-        val executionState = ExecutionBatchState(parameters.executionContext.getDocumentHeight())
-        executions.computeIfAbsent(parameters.executionContext.executionInput) {
-            executionState
-        }
-        return SimpleInstrumentationContext.noOp()
+    ): InstrumentationContext<ExecutionResult>? {
+        executions[parameters.executionContext.executionInput] = ExecutionBatchState(parameters.executionContext.getDocumentHeight())
```

Comment:
should this be

```kotlin
executions.computeIfAbsent(parameters.executionContext.executionInput) {
    ExecutionBatchState(parameters.executionContext.getDocumentHeight())
}
```

Reply:
I don't think we need to lock the map using `computeIfAbsent` we are already using a `ConcurrentHashMap` so we should be fine, same with the other instrumentation
https://github.com/ExpediaGroup/graphql-kotlin/blob/master/executions/graphql-kotlin-dataloader-instrumentation/src/main/kotlin/com/expediagroup/graphql/dataloader/instrumentation/syncexhaustion/state/SyncExecutionExhaustedState.kt#L55

Reply:
ill add it just in case to prevent any possible race condition